File: src/app/model/category.ts
import { Restaurant } from "./retaurant";
import { User } from "./user";


export interface Category{
  id:number,
  name: string,
  description:string,
  created:Date,
  updated:Date,
  deleted:boolean,
  categoryType:{
      id:number;
      name: string;
      description:string;
      created:Date;
      updated:Date;
      deleted:boolean;
  }
  restaurant:{
      id:number;
      name: string;
      description:string;
      phone:string;
      address:string;
      created:Date;
      updated:Date;
      deleted:boolean;
  }
  createdUser:User["createdUser"],
 updatedUser:User["updatedUser"],
}

File: QuanlyNhahang_API/Model/Category.cs
﻿using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace QuanlyNhahang_API.Model
{
    public class Category
    {
        [Key]
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Created { get; set; }
        public DateTime Updated { get; set; }
        public bool Deleted { get; set; }
        public virtual Restaurant Restaurant { get; set; }
        public virtual CategoryType CategoryType { get; set; }
        public virtual User CreatedUser { get; set; }
        public virtual User UpdatedUser { get; set; }
    }
}